rotkehlchen/tests/unit/decoders/test_metamask.py

```diff
@@ -0,0 +1,727 @@
+from rotkehlchen.chain.optimism.modules.metamask.constants import (
+    METAMASK_ROUTER as METAMASK_ROUTER_OPT,
+)
+from rotkehlchen.chain.polygon_pos.modules.metamask.constants import (
+    METAMASK_ROUTER as METAMASK_ROUTER_MATIC,
+)
+from rotkehlchen.constants.assets import A_AAVE, A_ETH, A_POLYGON_POS_MATIC, A_USDC, A_USDT
+from rotkehlchen.fval import FVal
+from rotkehlchen.history.events.structures.evm_event import EvmEvent
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.tests.utils.constants import A_OPTIMISM_USDT
+from rotkehlchen.tests.utils.ethereum import get_decoded_events_of_transaction
+from rotkehlchen.types import Location, TimestampMS, deserialize_evm_tx_hash
+
+A_LUX = Asset('eip155:1/erc20:0x88dafebb769311d7fbbeb9a21431fa026d4100d0')
+A_OTACON = Asset('eip155:1/erc20:0x0F17eeCcc84739b9450C88dE0429020e2DEC05eb')
+A_INU = Asset('eip155:1/erc20:0xc76d53f988820fe70e01eccb0248b312c2f1c7ca')
+A_MOG = Asset('eip155:1/erc20:0xaaeE1A9723aaDB7afA2810263653A34bA2C21C7a')
+A_INJ = Asset('eip155:1/erc20:0xe28b3B32B6c345A34Ff64674606124Dd5Aceca30')
+A_ARBITRUM_USDC = Asset('eip155:42161/erc20:0xaf88d065e77c8cC2239327C5EDb3A432268e5831')
+A_OPTIMISM_USDC = Asset('eip155:10/erc20:0x0b2C639c533813f4Aa9D7837CAf62653d097Ff85')
+A_POLYGON_USDC = Asset('eip155:137/erc20:0x2791Bca1f2de4661ED88A30C99A7a9449Aa84174')
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('ethereum_accounts', [['0x7e6Ee5d52825B3A3d9C500E7B8b0a2BAa1c91545']])
+def test_metamask_swap_token_to_eth(database, ethereum_inquirer, ethereum_accounts):
+    """Data taken from
+    https://etherscan.io/tx/0x9b675024d8648c3b590eff411fcf75a1199d10d1a3fe2ddbe50e166ce8b87cc9
+    """
+    tx_hash = deserialize_evm_tx_hash(
+        '0x9b675024d8648c3b590eff411fcf75a1199d10d1a3fe2ddbe50e166ce8b87cc9',
+    )
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=ethereum_inquirer,
+        database=database,
+        tx_hash=tx_hash,
+    )
+    metamask_router_address = string_to_evm_address(METAMASK_ROUTER_ETH)
+    user_address = ethereum_accounts[0]
+    timestamp = TimestampMS(1696160411000)
+    approval_amount = '115792089237316195423570985000000000000000000000000000000000'
+    swap_amount = '6550.721365244578638016'
+    received_amount = '0.017595546435556104'
+    gas_fees = '0.001533786820220988'
+    metamask_fee = '0.000153961031311116'
+    expected_events = [
+        EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal(gas_fees)),
+            location_label=user_address,
+            notes=f'Burned {gas_fees} ETH for gas',
+            counterparty=CPT_GAS,
+        ),
+        EvmEvent(
```

Comment:
more compact, here and in all other tests

```suggestion
        ), EvmEvent(
```